Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp

```diff
@@ -286,9 +337,19 @@ namespace Terrain
                     terrainExists = false;
                 }
 
+                // Find the best surface tag at this point.
+                AzFramework::SurfaceData::SurfaceTagWeight surfaceWeight;
+                AzFramework::Terrain::TerrainDataRequestBus::BroadcastResult(
+                    surfaceWeight, &AzFramework::Terrain::TerrainDataRequests::GetMaxSurfaceWeightFromFloats, x, y,
+                    AzFramework::Terrain::TerrainDataRequests::Sampler::DEFAULT, nullptr);
+                    
                 Physics::HeightMaterialPoint point;
                 point.m_height = height - worldCenterZ;
                 point.m_quadMeshType = terrainExists ? Physics::QuadMeshType::SubdivideUpperLeftToBottomRight : Physics::QuadMeshType::Hole;
+                point.m_materialIndex = FindSurfaceTagIndex(surfaceWeight.m_surfaceType);
+                
+                AZ_Warning("TerrainPhysicsCollider", point.m_materialIndex != InvalidSurfaceTagIndex, "SurfaceTag not assigned.");
+
```

Comment:
No need to warn on this, it's valid to have a surface tag that isn't mapped.  However, to deal with this, it looks like we'll need to do another task that was on our "to-do" list - add a default physics material to the component as well.  FindSurfaceTagIndex() would then return the default material index if there wasn't a specific tag mapping set up.